src/rendererbase.h

```diff
@@ -120,9 +120,9 @@ class RendererBase : public apf::MimoProcessor<Derived
         , reference_offset_orientation(fifo)
         , master_volume(fifo, 1)
         , processing(fifo, true)
-        , decay_exponent(fifo, params.get("decay_exponent", 1))
+        , decay_exponent(fifo, params.get("decay_exponent", 1.f))
```

Comment:
Wow, I think this is actually a bug, isn't it?

But your fix also has a bug: `decay_exponent` uses `sample_type`, not `float`. Sure, most of the time it's the same, but the whole point of `sample_type` is that it should also work with `double` (which it practically probably doesn't, but that's another story) or probably even some other type.

The following could work?

```c++
decay_exponent(fifo, params.get<sample_type>("decay_exponent", 1))
```

Reply:
Good point.
It produced just a warning. Doesn't this simply get promoted from int to float in this case?
However, this is later used with `params.get<float>("decay_exponent")`
so it should still be a float type, right?

> decay_exponent(fifo, params.get<sample_type>("decay_exponent", 1.f))

and 
> params.get("amplitude_reference_distance", 3.f))

BTW: controller.h:862 says it throws an error. Maybe that's why?

Reply:
OK, I had another look and this indeed is a bug!

We didn't notice it before because both `load_scene()` and `_create_spontaneous_scene()` (in `controller.h`) overwrite this value again (and they use the correct invocation of the `get()` function).
So this bug would only have an observable effect if no scene and no sound file is loaded.

> Doesn't this simply get promoted from int to float in this case?

No!
It turns out this is actually quite dangerous ...
The `get()` function is a function template and the target type of the conversion is in this case deduced from the default argument, so it was `int`.
So if the user originally specified `0.5` or `0.0`, this wouldn't be convertible and it would return the `int` value 1 (which would later be converted to a floating-point value).
However, if the user specified `0` on the command line or in the config file (which is a very likely value) everything would have worked fine.

So the best solution is to explicitly specify the template parameter.
After that, it doesn't matter anymore if `1` or `1.0f` or `1.0` is used as default value, because in this case it is indeed automatically converted to the given type.
I would prefer `1`, because there is no need to be more specific.
Or does this produce a warning in your case?
If yes, can you please show it?

Reply:
Ok, now I get it.